src/components/AuthModal/AuthModal.jsx

```diff
@@ -0,0 +1,287 @@
+import { useState } from "react";
+import { useAuth } from "../../context";
+import { loginService, signupService } from "../../services";
+import { useNavigate } from "react-router";
+import { ICONS_URL } from "../../constants";
+import styles from "./AuthModal.module.css";
+
+const dummyUser = {
+  email: "bablutailor@gmail.com",
+  password: "babluTailor123",
+};
+
+const AuthModal = ({ authModalOpen, setAuthModalOpen }) => {
+  const navigate = useNavigate();
+  const { authState, authDispatch } = useAuth();
+
+  const [login, setLogin] = useState({
+    email: "",
+    password: "",
+  });
+
+  const [signup, setSignup] = useState({
+    email: "",
+    password: "",
+    username: "",
+  });
+
+  const inputHandler = (e, action, inputType) => {
+    action === "login"
+      ? setLogin((prevState) => ({
+          ...prevState,
+          [inputType]: e.target.value,
+        }))
+      : setSignup((prevState) => ({
+          ...prevState,
+          [inputType]: e.target.value,
+        }));
+  };
+
+  const loginHandler = async (e) => {
+    try {
+      e.preventDefault();
+      const response = await loginService(login);
+      if (response.status === 200) {
+        localStorage.setItem("viskartToken", response.data.encodedToken);
+        localStorage.setItem(
+          "viskartUser",
+          JSON.stringify(response.data.foundUser)
+        );
+        authDispatch({
+          type: "LOGIN",
+          payload: {
+            token: response.data.encodedToken,
+            user: response.data.foundUser,
+          },
+        });
+        modalHandler("CLOSE");
+        alert("Login Successful");
+        navigate("/");
+      }
+      if (response.status === 404) {
+        throw new Error(
+          "The email entered is not Registered. Please Enter a valid Email"
+        );
+      } else if (response.status === 401) {
+        throw new Error("Incorrect Password! Please try again.");
+      }
+    } catch (error) {
+      alert(error);
+    }
+  };
+
+  const signUpHandler = async (e) => {
+    try {
+      e.preventDefault();
+      let response = await signupService(signup);
+      console.log("Signup: ", signup);
+      if (response.status === 201) {
+        authDispatch({
+          type: "SIGNUP",
+          payload: {
+            token: response.data.encodedToken,
+            user: response.data.createdUser,
+          },
+        });
+        console.log("User:", authState.user);
```

Comment:
You can remove this

Reply:
Sure will do it. Thank's for pointing it out 👍